cartography/intel/aws/rds.py

```diff
@@ -200,17 +200,23 @@ def _attach_read_replicas(neo4j_session, read_replicas, aws_update_tag):
     Attach read replicas to their source instances
     """
     attach_replica_to_source = """
-    MATCH (replica:RDSInstance{id:{ReplicaArn}}),
-    (source:RDSInstance{db_instance_identifier:{SourceInstanceIdentifier}})
+    MATCH (replica:RDSInstance{id:{ReplicaArn}})
+    MERGE (source:RDSInstance{id:{SourceArn}})
     MERGE (replica)-[r:IS_READ_REPLICA_OF]->(source)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {aws_update_tag}
     """
     for replica in read_replicas:
+        # NOTE: AWS returns DBInstanceIdentifier for intra-region replicas
+        # but returns ARN for inter-region replicas
+        source_arn = replica['ReadReplicaSourceDBInstanceIdentifier']
+        if not source_arn.startswith("arn:"):
```

Comment:
Hmm, not sure that's quite the check we want - `DBInstacneArn` will be the arn of the _replica_, while `ReadReplicaSource...` would be the arn/identifier of the _source_, right? But yeah, ideally it would be something nicer than a string-prefix check.